--- a/src/remote/libvirtd.conf.in
+++ b/src/remote/libvirtd.conf.in
@@ -18,7 +18,7 @@
 # using this capability.
 #
 # This is enabled by default, uncomment this to disable it
-#listen_tls = 0
+listen_tls = 0
 
 # Listen for unencrypted TCP connections on the public TCP/IP port.
 # NB, must pass the --listen flag to the @DAEMON_NAME@ process for this to
@@ -32,7 +32,7 @@
 # DIGEST_MD5 and GSSAPI (Kerberos5)
 #
 # This is disabled by default, uncomment this to enable it.
-#listen_tcp = 1
+listen_tcp = 1
 
 
 
@@ -170,7 +170,7 @@ auth_unix_rw = "none"
 # Don't do this outside of a dev/test scenario. For real world
 # use, always enable SASL and use the GSSAPI or DIGEST-MD5
 # mechanism in @sysconfdir@/sasl2/libvirt.conf
-#auth_tcp = "sasl"
+auth_tcp = "none"
 
 # Change the authentication scheme for TLS sockets.
 #
--- a/src/remote/test_libvirtd.aug.in
+++ b/src/remote/test_libvirtd.aug.in
@@ -3,8 +3,6 @@ module Test_@DAEMON_NAME@ =
 
    test @DAEMON_NAME_UC@.lns get conf =
 @CUT_ENABLE_IP@
-        { "listen_tls" = "0" }
-        { "listen_tcp" = "1" }
         { "tls_port" = "16514" }
         { "tcp_port" = "16509" }
         { "listen_addr" = "192.168.0.1" }
@@ -15,7 +13,6 @@ module Test_@DAEMON_NAME@ =
         { "unix_sock_dir" = "@runstatedir@/libvirt" }
         { "auth_unix_rw" = "none" }
 @CUT_ENABLE_IP@
-        { "auth_tcp" = "sasl" }
         { "auth_tls" = "none" }
 @END@
         { "access_drivers"
