--- a/src/qemu/qemu.conf
+++ b/src/qemu/qemu.conf
@@ -701,7 +701,7 @@
 # closed after keepalive_interval seconds of inactivity without
 # sending any keepalive messages.
 #
-#keepalive_interval = 5
+keepalive_interval = -1
 #keepalive_count = 5
 
 
--- a/src/qemu/test_libvirtd_qemu.aug.in
+++ b/src/qemu/test_libvirtd_qemu.aug.in
@@ -81,7 +81,6 @@ module Test_libvirtd_qemu =
 { "relaxed_acs_check" = "1" }
 { "lock_manager" = "lockd" }
 { "max_queued" = "0" }
-{ "keepalive_interval" = "5" }
 { "keepalive_count" = "5" }
 { "seccomp_sandbox" = "1" }
 { "migration_address" = "0.0.0.0" }
--- a/src/remote/libvirtd.conf.in
+++ b/src/remote/libvirtd.conf.in
@@ -467,8 +467,7 @@ auth_unix_rw = "none"
 # closed after keepalive_interval seconds of inactivity without
 # sending any keepalive messages.
 #
-#keepalive_interval = 5
-#keepalive_count = 5
+keepalive_interval = -1
 
 #
 # These configuration options are no longer used.  There is no way to
--- a/src/remote/test_libvirtd.aug.in
+++ b/src/remote/test_libvirtd.aug.in
@@ -50,8 +50,6 @@ module Test_libvirtd =
         { "audit_logging" = "1" }
         { "host_uuid" = "00000000-0000-0000-0000-000000000000" }
         { "host_uuid_source" = "smbios" }
-        { "keepalive_interval" = "5" }
-        { "keepalive_count" = "5" }
         { "keepalive_required" = "1" }
         { "admin_keepalive_required" = "1" }
         { "admin_keepalive_interval" = "5" }
